build: Improve depends build system robustness

hebasto opened the pull request:
The depends build system looks fragile in its current state. For example, on master (346e780442f91fc155dcc9c44eedf23ac0bb15a7):
```
$ make --no-print-directory -C depends print-qt_suffix
qt_suffix=everywhere-opensource-src-4.3.4.tar.xz
```

With this PR:
```
$ make --no-print-directory -C depends print-qt_suffix
qt_suffix=everywhere-opensource-src-5.15.3.tar.xz
```

Required for bitcoin/bitcoin#22555.

hebasto commented:
cc @dongcarl @fanquake @theuni

MarcoFalke commented:
For the scripted-diff please use `git ls-files ...` or similar. Otherwise it will change out-of-tree files as well.

hebasto commented:
Updated fdaeb5207bc58a7c93472d1bac82b2c373d6f1be -> b48fd7bf3f3276f5e21cd7de960a859754b20285 ([pr22552.01](https://github.com/hebasto/bitcoin/commits/pr22552.01) -> [pr22552.02](https://github.com/hebasto/bitcoin/commits/pr22552.02), [diff](https://github.com/hebasto/bitcoin/compare/pr22552.01..pr22552.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/22552#issuecomment-886535720)

jarolrod reviewed (commented):
Concept ACK

I agree with the vision this PR is trying to set. I wanted to try to document what this seems to be doing and aims to accomplish.

**What this PR is doing:**
We can notice some "fragility" in our build system when looking at the recent `Android` related issues and their patches. The OP shows a good example; somehow, the boost version ended up there. Another example is the patch set of #22555; it seems as though the wrong value is evaluated with specific variables.

Why might this be happening? This PR hits the nail on the head by attacking the use of recursive variables. Recursive variables make it quite hard to track where the value is coming from (to a build noob like me). The main change this PR is performing is refactoring from recursively expanded variables to simple expanded ones. This irons out any kinks that currently exist and seems to work for the Android build issues as reported here: https://github.com/bitcoin/bitcoin/pull/22555#issuecomment-886947551

Per the [make manual](https://www.gnu.org/software/make/manual/html_node/Flavors.html)
> "To avoid all the problems and inconveniences of recursively expanded variables, there is another flavor: simply expanded variables."

**Should we do this?**
Warning: The following opinions are from a build noob

We do not *need* to refactor to simply expanded variables. We could spend time figuring out what about our current setup and the use of recursive variables is contributing to the example in the OP and partially leading to other examples like #22522. 

Nevertheless, we should move to simply expanded variables. The clear benefit is, again, stated in the [make manual](https://www.gnu.org/software/make/manual/html_node/Flavors.html):
> "Simply expanded variables generally make complicated makefile programming more predictable because they work like variables in most programming languages."

Performing this refactor makes future changes more accessible because we can easily track where values come from. Also, and while not a directly supporting reason for this change, it can lower the entry and learning curve barrier to contributing to our build system.

Also, simply expanded variables are newer. Newer = Better ☺️

**Notes on commits:**
- dc54e4ad2c58cff78ad618f1c577aae704bc7901
  - simlar to explanation for 2ee0cf35d09a4b7a08361ecf2576e0e15e93f5f2
- c705d1ba86938387b9f4bb2e4b3eea6c8b2b44fe
  - This is mainly a cosmetic change. Not required
- 2ee0cf35d09a4b7a08361ecf2576e0e15e93f5f2
  - If we are going to refactor to simple variables, this is needed to keep the proper order of evaluation. X.mk depends on native_X.mk.
- 096d24f3ce24fcc38a1dfa85e5a08dd7ca494e75
  - This is a cosmetic change
- 4e70c20acdefc9e596297b322e9f2897f9dc4654 & 638133188fa21f17e65e73f08c0ff7b87a58f446
  - performs migration from recursive to simple variables

jarolrod commented:
Guix Hashes, mine match @hebasto:

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

4ea194b0f639a35c111de60d646ec7adcde82a5ab0c2b4d82637046909c9967c  guix-build-638133188fa2/output/aarch64-linux-gnu/SHA256SUMS.part
38178d82ffb02f3b776a570bddc4331078b27a22a5a6f6f21eeec95fee556290  guix-build-638133188fa2/output/aarch64-linux-gnu/bitcoin-638133188fa2-aarch64-linux-gnu-debug.tar.gz
085c15770e12d60ba143d7795732931813255843af3cf308f18aafa47650e949  guix-build-638133188fa2/output/aarch64-linux-gnu/bitcoin-638133188fa2-aarch64-linux-gnu.tar.gz
0a0f49b35abc654842d988bfa8ccd66f104eff223baded0c0c48fe306bfc3189  guix-build-638133188fa2/output/arm-linux-gnueabihf/SHA256SUMS.part
a3ef166964fcecbe12f24559b7de2be2d535f4c4d1480a14d230c7027183bbcb  guix-build-638133188fa2/output/arm-linux-gnueabihf/bitcoin-638133188fa2-arm-linux-gnueabihf-debug.tar.gz
fcf12dc881680552a262754f520c52b30831aeeef4c4e3163d0a00b9b7dd8494  guix-build-638133188fa2/output/arm-linux-gnueabihf/bitcoin-638133188fa2-arm-linux-gnueabihf.tar.gz
12c4769900aca1fd2133f45c3cf5b84d6bc3e28c55f3e5349e65eabcc2d4d606  guix-build-638133188fa2/output/dist-archive/bitcoin-638133188fa2.tar.gz
78b857871eb735a1add4018ff38a0b13bb6c6eae49d22394dbe8c039b6d8f728  guix-build-638133188fa2/output/powerpc64-linux-gnu/SHA256SUMS.part
cef24717d877e2a069fb2578a5675dba2adb0ab37a8c735fd1c42f03b67da744  guix-build-638133188fa2/output/powerpc64-linux-gnu/bitcoin-638133188fa2-powerpc64-linux-gnu-debug.tar.gz
465f37bcbb72c03795085e2e47c647ac8d9db9894e19edabaa265b17faec790c  guix-build-638133188fa2/output/powerpc64-linux-gnu/bitcoin-638133188fa2-powerpc64-linux-gnu.tar.gz
28ee721dd23c5dffb993827dc319d91424925eb4427443e9ed0a65f2b89bd1ec  guix-build-638133188fa2/output/powerpc64le-linux-gnu/SHA256SUMS.part
c57922fb4e32f62ac0e80383669b350e2c1973c3c17003a39a65dc15790c11af  guix-build-638133188fa2/output/powerpc64le-linux-gnu/bitcoin-638133188fa2-powerpc64le-linux-gnu-debug.tar.gz
38cd9bdc692390a166ec2926eea66c80c3ceb09eda49d251baf4b028717de1ae  guix-build-638133188fa2/output/powerpc64le-linux-gnu/bitcoin-638133188fa2-powerpc64le-linux-gnu.tar.gz
bb835248e4212751b51d28e1105347c129b5bc9be9675adc4a05107b97d04cff  guix-build-638133188fa2/output/riscv64-linux-gnu/SHA256SUMS.part
3249955ece9fbc54c15bc3a450d47a9908578f0de1d94e148874926cdf378781  guix-build-638133188fa2/output/riscv64-linux-gnu/bitcoin-638133188fa2-riscv64-linux-gnu-debug.tar.gz
ef21bd74300b777537b0e0cf836df06f231cad38eaf31032d2a0688975cd878e  guix-build-638133188fa2/output/riscv64-linux-gnu/bitcoin-638133188fa2-riscv64-linux-gnu.tar.gz
367ff0943f3a662d4ccb8e2351ac6bda96438d26e602814b4625b9d19e74dfc0  guix-build-638133188fa2/output/x86_64-apple-darwin18/SHA256SUMS.part
4c8eadf70973f1b01037e4717e3152ca704af6591d2def66cdc0d9e997cce4e5  guix-build-638133188fa2/output/x86_64-apple-darwin18/bitcoin-638133188fa2-osx-unsigned.dmg
d1d844548fccac9de6d229a93b521a8cf88265dc9f034cd4a109dbaec840d4aa  guix-build-638133188fa2/output/x86_64-apple-darwin18/bitcoin-638133188fa2-osx-unsigned.tar.gz
0dcb20052e0196c58eeb66f56fe15a5ab6466376203919e01348bfde6718f3f9  guix-build-638133188fa2/output/x86_64-apple-darwin18/bitcoin-638133188fa2-osx64.tar.gz
4aded0867bcc8d163f9137f131d82bc157a83043c7907474d76ddd4b3f18c40d  guix-build-638133188fa2/output/x86_64-linux-gnu/SHA256SUMS.part
d3031ec2e29a888e4462a44e92ffa77feedd8ac243da682071fce419cf7e5f26  guix-build-638133188fa2/output/x86_64-linux-gnu/bitcoin-638133188fa2-x86_64-linux-gnu-debug.tar.gz
a6722dca18f7c400c53d3576b6d438bc397d084f200eff5c32e7b77c8aa7a8a5  guix-build-638133188fa2/output/x86_64-linux-gnu/bitcoin-638133188fa2-x86_64-linux-gnu.tar.gz
0285facc842ecd4b492c3b62a21079e83b0e81bc0a01ee750543ba0c341a74bf  guix-build-638133188fa2/output/x86_64-w64-mingw32/SHA256SUMS.part
3ec91feb1265d8492ef10d933ed18ee54fe0baf42abd03232fb747cce0a9c925  guix-build-638133188fa2/output/x86_64-w64-mingw32/bitcoin-638133188fa2-win-unsigned.tar.gz
5bfdb18903f4076f7484bbd82079c94eb8ccaf5bb5523a5a59abffbdaa686194  guix-build-638133188fa2/output/x86_64-w64-mingw32/bitcoin-638133188fa2-win64-debug.zip
8d8dc8b4a0d1f5e49189ff74b35803bd228bc8b560fd2bbf84fc5e94a1bb6d16  guix-build-638133188fa2/output/x86_64-w64-mingw32/bitcoin-638133188fa2-win64-setup-unsigned.exe
67d4c46850222e57d688beed894c94d84a88219f0f9a2ef6d2950d3a3aa1df9d  guix-build-638133188fa2/output/x86_64-w64-mingw32/bitcoin-638133188fa2-win64.zip
```

laanwj commented:
> Why might this be happening? This PR hits the nail on the head by attacking the use of recursive variables.

Thanks @jarolrod for your detailed explanation, it wasn't kind of clear to me in the OP what was exactly fragile here, and how it is fixed. I don't know a lot about makefile internals, but avoiding inadvertent issues that could result in wrong builds seems important. Concept ACK.

Would be good if @theuni has a look over the changes I think.

hebasto commented:
Rebased ad2a195b616e1822cdd6d47c792cbf21a11df7c5 -> f2812ac19667015423cdcea884745e8b79520dc7 ([pr22552.04](https://github.com/hebasto/bitcoin/commits/pr22552.04) -> [pr22552.05](https://github.com/hebasto/bitcoin/commits/pr22552.05)) due to the conflict with #23494.

hebasto commented:
@theuni 

Would you mind having a look into this PR?

hebasto commented:
Rebased f2812ac19667015423cdcea884745e8b79520dc7 -> 61b03345460a6d249af7e8829455d20786764c7a ([pr22552.05](https://github.com/hebasto/bitcoin/commits/pr22552.05) -> [pr22552.06](https://github.com/hebasto/bitcoin/commits/pr22552.06)) due to the conflict with #23535.

hebasto commented:
Rebased 61b03345460a6d249af7e8829455d20786764c7a -> 01938b9b32ee6915890ad0ae5f26a3d59310f82f ([pr22552.06](https://github.com/hebasto/bitcoin/commits/pr22552.06) -> [pr22552.07](https://github.com/hebasto/bitcoin/commits/pr22552.07)) due to the conflict with #23580.

hebasto commented:
Rebased 01938b9b32ee6915890ad0ae5f26a3d59310f82f -> a3a4c46f36cebe57f3eacfdf5335d5e766a8c86c ([pr22552.07](https://github.com/hebasto/bitcoin/commits/pr22552.07) -> [pr22552.08](https://github.com/hebasto/bitcoin/commits/pr22552.08)) due to the conflict with #23618.

hebasto commented:
Rebased a3a4c46f36cebe57f3eacfdf5335d5e766a8c86c -> 2c586571bc7c0877a4e76906cdd752cba1d9a2ec ([pr22552.08](https://github.com/hebasto/bitcoin/commits/pr22552.08) -> [pr22552.09](https://github.com/hebasto/bitcoin/commits/pr22552.09)) due to the conflict with #23489.

hebasto commented:
Rebased 2c586571bc7c0877a4e76906cdd752cba1d9a2ec -> fe44782aec88f59f75b1461904a1be1c7eaaa7bc ([pr22552.09](https://github.com/hebasto/bitcoin/commits/pr22552.09) -> [pr22552.10](https://github.com/hebasto/bitcoin/commits/pr22552.10)) due to the conflicts with #23495 and #23673.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
45c47261818444ffc277421065cc20eca1f4ca031a861e5b9c0c2393ac3aa003  guix-build-fe44782aec88/output/aarch64-linux-gnu/SHA256SUMS.part
4657c1fe8222e25e5fe9922eac104bbb8b121c2b7e317485cd15eb9b57745217  guix-build-fe44782aec88/output/aarch64-linux-gnu/bitcoin-fe44782aec88-aarch64-linux-gnu-debug.tar.gz
b2e80d19a297d89e04ae7c47e8064af463c3f531bd180350f58b5577dd2bd44f  guix-build-fe44782aec88/output/aarch64-linux-gnu/bitcoin-fe44782aec88-aarch64-linux-gnu.tar.gz
48fe605a1604f74d94e216024aa2082553c2be3ce355ea6e9b8c6880aaae455f  guix-build-fe44782aec88/output/arm-linux-gnueabihf/SHA256SUMS.part
328692804231c42e0e51c80f884943297d5ae243f7272cb5954a86f620303779  guix-build-fe44782aec88/output/arm-linux-gnueabihf/bitcoin-fe44782aec88-arm-linux-gnueabihf-debug.tar.gz
103feb1b9e98ef8022ff766f2d5f9b1fb78353900a4074ff758f999fc1c29691  guix-build-fe44782aec88/output/arm-linux-gnueabihf/bitcoin-fe44782aec88-arm-linux-gnueabihf.tar.gz
588d1a39c94a9f7719299168ec701fee46e99485069d9afee2fec01342a4a23c  guix-build-fe44782aec88/output/dist-archive/bitcoin-fe44782aec88.tar.gz
3000306671716e91d2abe4be0f37eabba6ebef88b1f0e6712e6785ae31336d9d  guix-build-fe44782aec88/output/powerpc64-linux-gnu/SHA256SUMS.part
3f66fcb2f94604d5ddef11416ef3135cd30ab5386c5088aa9f4af85b7e67c84d  guix-build-fe44782aec88/output/powerpc64-linux-gnu/bitcoin-fe44782aec88-powerpc64-linux-gnu-debug.tar.gz
a1946293419cdcbd08ae4e2bb98e7713992ca3577e55dbc184fad5e288eb10c9  guix-build-fe44782aec88/output/powerpc64-linux-gnu/bitcoin-fe44782aec88-powerpc64-linux-gnu.tar.gz
b14aaddeb95c11a274bac73d9d4b614a3c8356a79ebccfe81bc867b0a48d0893  guix-build-fe44782aec88/output/powerpc64le-linux-gnu/SHA256SUMS.part
2530561fcb1ab7e315db28257c34205c8b72bed88b6d0f9180250331d98159ce  guix-build-fe44782aec88/output/powerpc64le-linux-gnu/bitcoin-fe44782aec88-powerpc64le-linux-gnu-debug.tar.gz
83d5df89fcc4c33a86c6141c11184944e90f71464b0556939921df0765fa45f2  guix-build-fe44782aec88/output/powerpc64le-linux-gnu/bitcoin-fe44782aec88-powerpc64le-linux-gnu.tar.gz
258b9bb7035d0aca2ee956d24e37e90f99be25acd99815d121162266824e3ef9  guix-build-fe44782aec88/output/riscv64-linux-gnu/SHA256SUMS.part
93060d05982d8a90c0387d6a8be76cc067592575bce15680540ce3eea5973036  guix-build-fe44782aec88/output/riscv64-linux-gnu/bitcoin-fe44782aec88-riscv64-linux-gnu-debug.tar.gz
f6f677139d1bef31ef97d7022b07643a6d203726f683840b47f705594515ca36  guix-build-fe44782aec88/output/riscv64-linux-gnu/bitcoin-fe44782aec88-riscv64-linux-gnu.tar.gz
da03c88523d6697dd557ea7a08cc9e06c9947eb65ef23f5c2141abe868457dc8  guix-build-fe44782aec88/output/x86_64-apple-darwin/SHA256SUMS.part
e9d3a25b5c2daef51a0860fa62ce45d865247b85d527c744beba89ac57797e0c  guix-build-fe44782aec88/output/x86_64-apple-darwin/bitcoin-fe44782aec88-osx-unsigned.dmg
8365d8169bc1eefde3c1a531e652e4f7b151283c139618e440a380d8cee5f09c  guix-build-fe44782aec88/output/x86_64-apple-darwin/bitcoin-fe44782aec88-osx-unsigned.tar.gz
662d87b608b304b57db77e2b57115b1d6a8556e63a2326331e9c27e10ff7aba2  guix-build-fe44782aec88/output/x86_64-apple-darwin/bitcoin-fe44782aec88-osx64.tar.gz
eea417b76b9db47f0a9d57a84ecb72487e847dd0cc01786ed6db65bcc0219f8e  guix-build-fe44782aec88/output/x86_64-linux-gnu/SHA256SUMS.part
49345d741c65ccdb943d69497e3033d194dae98b21081e0694847eed225651c9  guix-build-fe44782aec88/output/x86_64-linux-gnu/bitcoin-fe44782aec88-x86_64-linux-gnu-debug.tar.gz
4d821b0fe896f51f57336102e4b171e8b04cd7ea873c17d7632e85613074af67  guix-build-fe44782aec88/output/x86_64-linux-gnu/bitcoin-fe44782aec88-x86_64-linux-gnu.tar.gz
1f69e56deba9c44ebef21c89c70e0032116bcc0fdcd5ec12494c7147f73ba89b  guix-build-fe44782aec88/output/x86_64-w64-mingw32/SHA256SUMS.part
ee073d1b4e6512db9d8cae43f15b45e87df8d06cf6e651781bc6aa5b2cd8b659  guix-build-fe44782aec88/output/x86_64-w64-mingw32/bitcoin-fe44782aec88-win-unsigned.tar.gz
b444c90a248299f7ed0ef9050bc97e85601a525c83779f1e40650b2d1cd7b3ac  guix-build-fe44782aec88/output/x86_64-w64-mingw32/bitcoin-fe44782aec88-win64-debug.zip
d90589dc55e6da2b15c268d758c4b832bdf9a3e2c11c678e334efcd8fe70e2cb  guix-build-fe44782aec88/output/x86_64-w64-mingw32/bitcoin-fe44782aec88-win64-setup-unsigned.exe
e0152f77cf341a04eab49cc325e14c219fafa955dc4b62b341c6239885721a1b  guix-build-fe44782aec88/output/x86_64-w64-mingw32/bitcoin-fe44782aec88-win64.zip
```

hebasto commented:
Rebased fe44782aec88f59f75b1461904a1be1c7eaaa7bc -> 77ec0f5ec7eff638b49d0915796a7ada53bf8ebd ([pr22552.10](https://github.com/hebasto/bitcoin/commits/pr22552.10) -> [pr22552.11](https://github.com/hebasto/bitcoin/commits/pr22552.11)) due to the conflict with #23724.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
70c45fec102269ccc1914d1e01d65b1f9efa04dfe2a490ad55f10fda898dde34  guix-build-77ec0f5ec7ef/output/aarch64-linux-gnu/SHA256SUMS.part
f0c2bce986ef168ef2284bfec03912ea8e7f8abe38c6978f056f8c13547d434e  guix-build-77ec0f5ec7ef/output/aarch64-linux-gnu/bitcoin-77ec0f5ec7ef-aarch64-linux-gnu-debug.tar.gz
b89adc7c8f61c370a3474e8d6637dd00196d3a545448067f6b33e58a9af334e3  guix-build-77ec0f5ec7ef/output/aarch64-linux-gnu/bitcoin-77ec0f5ec7ef-aarch64-linux-gnu.tar.gz
9f56d4fd9d650c60d5d2208bdf7ee42b8d4a47187e9d3335030f2b097a33f342  guix-build-77ec0f5ec7ef/output/arm-linux-gnueabihf/SHA256SUMS.part
e6afd1974ed9bdc0f0f4b71134cdfbe60d7c361c33ebe60718c9faa4286753bc  guix-build-77ec0f5ec7ef/output/arm-linux-gnueabihf/bitcoin-77ec0f5ec7ef-arm-linux-gnueabihf-debug.tar.gz
21811f544088589507d8174314c5dc00ec40749e2620552bea76753756f680df  guix-build-77ec0f5ec7ef/output/arm-linux-gnueabihf/bitcoin-77ec0f5ec7ef-arm-linux-gnueabihf.tar.gz
bcca8bae219288d47bb57c8d24e8298ff10b6c7a0ba6eef7ca77324f5c606bd9  guix-build-77ec0f5ec7ef/output/dist-archive/bitcoin-77ec0f5ec7ef.tar.gz
df936bdf6c71147750c37f8af942dc251b14fc8e68dfb64b6fad30d85d000a84  guix-build-77ec0f5ec7ef/output/powerpc64-linux-gnu/SHA256SUMS.part
076d645e33d745edff4a89fbfb0bb4f7b42c6c2fc53a7073e94bdb69d63c3fa3  guix-build-77ec0f5ec7ef/output/powerpc64-linux-gnu/bitcoin-77ec0f5ec7ef-powerpc64-linux-gnu-debug.tar.gz
fd9d0c6b1ba81570f411c5f11c552142f29815cfe48a101eae24ecc04a5be622  guix-build-77ec0f5ec7ef/output/powerpc64-linux-gnu/bitcoin-77ec0f5ec7ef-powerpc64-linux-gnu.tar.gz
ee9c34b1b5a510a125a9d05c0e49af628432a7e5787ed945dafcba4f1f1cad21  guix-build-77ec0f5ec7ef/output/powerpc64le-linux-gnu/SHA256SUMS.part
5e501ec61a864b344d0a71db6dd56d2b2285d55ad893c9d2ec91e72a37d027e8  guix-build-77ec0f5ec7ef/output/powerpc64le-linux-gnu/bitcoin-77ec0f5ec7ef-powerpc64le-linux-gnu-debug.tar.gz
af1fdac65b55d1696b66f6c075ec127645d31a79c758d7b22f9cde60b0141609  guix-build-77ec0f5ec7ef/output/powerpc64le-linux-gnu/bitcoin-77ec0f5ec7ef-powerpc64le-linux-gnu.tar.gz
28b7257adef047ddc761ac69a2f7bacc8eccfd630398b79449bcec7d4096de53  guix-build-77ec0f5ec7ef/output/riscv64-linux-gnu/SHA256SUMS.part
c5ff4288d5f99ea1157330e7fed4ff8aa1b03c5cec8a69f13821abd0aa460ce8  guix-build-77ec0f5ec7ef/output/riscv64-linux-gnu/bitcoin-77ec0f5ec7ef-riscv64-linux-gnu-debug.tar.gz
2f467b36305488a691c3871bb004e3f999f2395c04538c2bbbd43a91dcbe7675  guix-build-77ec0f5ec7ef/output/riscv64-linux-gnu/bitcoin-77ec0f5ec7ef-riscv64-linux-gnu.tar.gz
33a87ae14b77518ea766042741e389fdb7d0b4f9b36a55273b733935fcb62320  guix-build-77ec0f5ec7ef/output/x86_64-apple-darwin/SHA256SUMS.part
d539ca75848ff0a06cd589d38ba0fa1073e1bd421e93a00c779a975f5209bfe8  guix-build-77ec0f5ec7ef/output/x86_64-apple-darwin/bitcoin-77ec0f5ec7ef-osx-unsigned.dmg
1e78e96f703d50264b283dce13bbea07dc426697c9f423f367c231dfe27a8ecd  guix-build-77ec0f5ec7ef/output/x86_64-apple-darwin/bitcoin-77ec0f5ec7ef-osx-unsigned.tar.gz
96a2a33c9a4ef733278087c29fc796a5d63b6b5005a0cfe2b9407570a9425bf2  guix-build-77ec0f5ec7ef/output/x86_64-apple-darwin/bitcoin-77ec0f5ec7ef-osx64.tar.gz
a576e9b4c4abd891a5d0e208c0a467549effebe4acb97fce69ed40b737043d0f  guix-build-77ec0f5ec7ef/output/x86_64-linux-gnu/SHA256SUMS.part
176fb8bfd5fc8782bf273fb3f8ec81a7a4c7624e6f24751d567d4aeaef92cccb  guix-build-77ec0f5ec7ef/output/x86_64-linux-gnu/bitcoin-77ec0f5ec7ef-x86_64-linux-gnu-debug.tar.gz
d6220d1f1029d641aa1e194d10c5c3e898b308019088a8a89624e2e4c19388a0  guix-build-77ec0f5ec7ef/output/x86_64-linux-gnu/bitcoin-77ec0f5ec7ef-x86_64-linux-gnu.tar.gz
d3545ff15199a3a354500b3b5bb35dd0506b188e72dc946a750420f0d54f9357  guix-build-77ec0f5ec7ef/output/x86_64-w64-mingw32/SHA256SUMS.part
1b0387e9fb9921fec13a56e369e3d56431497d89160003b7681bc08cbe371595  guix-build-77ec0f5ec7ef/output/x86_64-w64-mingw32/bitcoin-77ec0f5ec7ef-win-unsigned.tar.gz
553ee10401b00da12b5649a25561e9157cc7053066413e6f0c34cfe877950c1b  guix-build-77ec0f5ec7ef/output/x86_64-w64-mingw32/bitcoin-77ec0f5ec7ef-win64-debug.zip
926f94b1bc0d1b9489d96244de78da0bf2d103d54dda529d54211b3a8f74286b  guix-build-77ec0f5ec7ef/output/x86_64-w64-mingw32/bitcoin-77ec0f5ec7ef-win64-setup-unsigned.exe
ebf511d27ff1ed5b6fa7e42cb3fd07d5019c9ca8a8784c5647c9f86febea93ae  guix-build-77ec0f5ec7ef/output/x86_64-w64-mingw32/bitcoin-77ec0f5ec7ef-win64.zip
```

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
081859374c87b5ac084250396bcb72829398c964bdc5595f6499fa0c6458688b  guix-build-0931be380420/output/aarch64-linux-gnu/SHA256SUMS.part
f9e49ca60c7e7498a27307490a3d1f6046b35a9ce1cfe6ce0b73b72a3708cf15  guix-build-0931be380420/output/aarch64-linux-gnu/bitcoin-0931be380420-aarch64-linux-gnu-debug.tar.gz
52545dca83c84eb93ee63e02309ace5efe4d46ac640080542ba1f76a26b279c8  guix-build-0931be380420/output/aarch64-linux-gnu/bitcoin-0931be380420-aarch64-linux-gnu.tar.gz
e822107401b413cd54d63160d7086f76a7ba965ebfc1dbddabeeaa0e8458d70f  guix-build-0931be380420/output/arm-linux-gnueabihf/SHA256SUMS.part
6da545015ea27e17d46e82e7c067ecc774ccc84ecd26ffe90a211b5eb27eeda7  guix-build-0931be380420/output/arm-linux-gnueabihf/bitcoin-0931be380420-arm-linux-gnueabihf-debug.tar.gz
d2a17a3ee9722ef7cbf7998e973394f09c7737d52f773831c57d16d9b2168b86  guix-build-0931be380420/output/arm-linux-gnueabihf/bitcoin-0931be380420-arm-linux-gnueabihf.tar.gz
9287f42388ba71b0b47d1c2d110a344d30619e1452b958c8d2868664c072574f  guix-build-0931be380420/output/arm64-apple-darwin/SHA256SUMS.part
d67fecd159f6ef1dcb02b65501faf04827ea8495859953f76fae23659ef483a8  guix-build-0931be380420/output/arm64-apple-darwin/bitcoin-0931be380420-arm64-apple-darwin.tar.gz
fdcc8cc738b35597ac72900a9631011d031009ca6505ca18163f0f6a15af5751  guix-build-0931be380420/output/arm64-apple-darwin/bitcoin-0931be380420-osx-unsigned.dmg
43080ab435f87af236b2f90bfca2a84ab98ee835c0a0b2c7b7fa284d9f47bee0  guix-build-0931be380420/output/arm64-apple-darwin/bitcoin-0931be380420-osx-unsigned.tar.gz
4bd4927250c892c3cf7ea761440dd60ba1946bb7525c755735495d0d1b56cad5  guix-build-0931be380420/output/dist-archive/bitcoin-0931be380420.tar.gz
d25aae66166c3b9f50335331ae403afdae3d329c9dbe75128085c0db9d40e946  guix-build-0931be380420/output/powerpc64-linux-gnu/SHA256SUMS.part
6eeebb17df4508607f3db58e6defdc5ce0ace763ae17b21dec3f91ba411b35e2  guix-build-0931be380420/output/powerpc64-linux-gnu/bitcoin-0931be380420-powerpc64-linux-gnu-debug.tar.gz
2db6be9a29d715c1f6129320117bba0ce68b422e8d9c9ecc65f54ff61c0ee636  guix-build-0931be380420/output/powerpc64-linux-gnu/bitcoin-0931be380420-powerpc64-linux-gnu.tar.gz
d9f90143c045dbd4d756a1f3a79aa735aed7c0cc786d325d3c08cf08820f9f5b  guix-build-0931be380420/output/powerpc64le-linux-gnu/SHA256SUMS.part
baf5a43cf35ba2fafd1cbfbbc645fb5986795875c2a80202f9d02e0a5886376e  guix-build-0931be380420/output/powerpc64le-linux-gnu/bitcoin-0931be380420-powerpc64le-linux-gnu-debug.tar.gz
4844d2e06b82cd7216ff23c03ba166e02cd8be1239cb933a45d0f8e11b0ecb52  guix-build-0931be380420/output/powerpc64le-linux-gnu/bitcoin-0931be380420-powerpc64le-linux-gnu.tar.gz
3b03c76f8f00c04797d44317c509916894e008f53911eda830c66cd536c73bcd  guix-build-0931be380420/output/riscv64-linux-gnu/SHA256SUMS.part
9b21a5b3bba38dc8bcbca0fc04b74487b2659ccfc86acbf5346ce9175c6f43a6  guix-build-0931be380420/output/riscv64-linux-gnu/bitcoin-0931be380420-riscv64-linux-gnu-debug.tar.gz
675a966f84bc83e3e15ee7d2487a9849e23bc96527cebd37febf0cbe552de36d  guix-build-0931be380420/output/riscv64-linux-gnu/bitcoin-0931be380420-riscv64-linux-gnu.tar.gz
ff3430d94f88b13124041f1c2ae85a61428989dd7daa05948022c10febd82982  guix-build-0931be380420/output/x86_64-apple-darwin/SHA256SUMS.part
9375b5e3340ea63b4ec95607dbd5d803f80dcce1278fc3784eaee408ef1f04a3  guix-build-0931be380420/output/x86_64-apple-darwin/bitcoin-0931be380420-osx-unsigned.dmg
e49c3ca35e3bd3820805ab025a9b6f1a287643f024f69866384af3cbd085535d  guix-build-0931be380420/output/x86_64-apple-darwin/bitcoin-0931be380420-osx-unsigned.tar.gz
2d467b56cc7b466c2e387a5480f8e88430945a850414a1a49c95e484a64cbc31  guix-build-0931be380420/output/x86_64-apple-darwin/bitcoin-0931be380420-osx64.tar.gz
ccb502584fd99f7fbfa916549d3189ace6995d56f9462c55730108ef13246e56  guix-build-0931be380420/output/x86_64-linux-gnu/SHA256SUMS.part
fa225cdad9740c2a0966495c3eba0facbaa8c7fc66a0b7676768609844037bfa  guix-build-0931be380420/output/x86_64-linux-gnu/bitcoin-0931be380420-x86_64-linux-gnu-debug.tar.gz
56d91f6388e2da25a24eef3284e9ce57d0063ed7fcd81e6254bb0888eeac491e  guix-build-0931be380420/output/x86_64-linux-gnu/bitcoin-0931be380420-x86_64-linux-gnu.tar.gz
```

hebasto commented:
Rebased 0931be3804208bc7bdca036dc15abaf47f6ea963 -> 17ef4c9c5888bdc2be18184f1fcdae7c0aac1eb8 ([pr22552.12](https://github.com/hebasto/bitcoin/commits/pr22552.12) -> [pr22552.13](https://github.com/hebasto/bitcoin/commits/pr22552.13)) due to the conflict with #24288.

hebasto commented:
Rebased 17ef4c9c5888bdc2be18184f1fcdae7c0aac1eb8 -> 4cba82dcb232ab2f46703d94511904c13c9445be ([pr22552.13](https://github.com/hebasto/bitcoin/commits/pr22552.13) -> [pr22552.14](https://github.com/hebasto/bitcoin/commits/pr22552.14)):
- rebased
- PR description has been updated

hebasto commented:
Rebased 4cba82dcb232ab2f46703d94511904c13c9445be -> 83f7c336d16e644848ca60bde11014164c1beb0d ([pr22552.14](https://github.com/hebasto/bitcoin/commits/pr22552.14) -> [pr22552.15](https://github.com/hebasto/bitcoin/commits/pr22552.15)) due to the conflict with #25046.

hebasto committed:
scripted-diff: Make (package)_X variables simply expanded ones

-BEGIN VERIFY SCRIPT-
FILES=$(git ls-files depends/packages)
sed -i 's/package=/package := /' -- $FILES
sed -i -e 's/$(package)\(\w\+\)=/$(package)\1 := /' -- $FILES
-END VERIFY SCRIPT-

hebasto committed:
build: Make packages related variables simply expanded ones

hebasto commented:
Rebased 83f7c336d16e644848ca60bde11014164c1beb0d -> df6dccde8e1cfc498700b3e10ab3b873156a3aed ([pr22552.15](https://github.com/hebasto/bitcoin/commits/pr22552.15) -> [pr22552.16](https://github.com/hebasto/bitcoin/commits/pr22552.16)).

jarolrod commented:
GUIX hashes:

x86:
```

```

arm64:
```
$ env HOSTS='arm-linux-gnueabihf arm64-apple-darwin powerpc64-linux-gnu powerpc64le-linux-gnu riscv64-linux-gnu x86_64-apple-darwin x86_64-linux-gnu x86_64-w64-mingw32' ./contrib/guix/guix-build
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

c9f5832b2a7a9138650abd07cb99e61c5e4806b5f0a37632b5e9fcc4e2f2cb50  guix-build-df6dccde8e1c/output/arm-linux-gnueabihf/SHA256SUMS.part
dff03c057136f27718793a3586518599c5fde872506ce6b3d445e9e9975bb125  guix-build-df6dccde8e1c/output/arm-linux-gnueabihf/bitcoin-df6dccde8e1c-arm-linux-gnueabihf-debug.tar.gz
a73569a8aac5940f7b1bd683c2d5ffb3b899beeb29fe521351cb481b4897b747  guix-build-df6dccde8e1c/output/arm-linux-gnueabihf/bitcoin-df6dccde8e1c-arm-linux-gnueabihf.tar.gz
baafc34efccded2ae164b1b3b5ce3c7547322ba81498f7bdf9d163490d4a402e  guix-build-df6dccde8e1c/output/arm64-apple-darwin/SHA256SUMS.part
abb2cac3e4270a107359d3182401a1582f39ca55f8a95b00e441abf6738b3ff6  guix-build-df6dccde8e1c/output/arm64-apple-darwin/bitcoin-df6dccde8e1c-arm64-apple-darwin-unsigned.dmg
c545b8871acad16c3f4ea83e382be59101a1306ed5faf03c9d400311fa914a5a  guix-build-df6dccde8e1c/output/arm64-apple-darwin/bitcoin-df6dccde8e1c-arm64-apple-darwin-unsigned.tar.gz
d6c84396b152c21f94f174505d3b689f94726f7e2d750a9b5415a561f034c6ec  guix-build-df6dccde8e1c/output/arm64-apple-darwin/bitcoin-df6dccde8e1c-arm64-apple-darwin.tar.gz
7cdd953cb6334360e609f74030750ddd64b22fb7787a6faac95939eab8dbc112  guix-build-df6dccde8e1c/output/dist-archive/bitcoin-df6dccde8e1c.tar.gz
0a62c1dcf1e8789fb7051d86dc0ae576e5feb449a9f394224163e3476d73eab9  guix-build-df6dccde8e1c/output/powerpc64-linux-gnu/SHA256SUMS.part
b87fd60479ebb0f6c6659e7911c251dc576260146f3e6ab24df9e2b72f6e029e  guix-build-df6dccde8e1c/output/powerpc64-linux-gnu/bitcoin-df6dccde8e1c-powerpc64-linux-gnu-debug.tar.gz
d18aa375d1bf87e682e10d78621f8ce72306f27c098250de806e6d7aa11b574f  guix-build-df6dccde8e1c/output/powerpc64-linux-gnu/bitcoin-df6dccde8e1c-powerpc64-linux-gnu.tar.gz
827176015b3c1ee7a04f62f189145efe804c5b506cf1c02fac77aff1d695b4ef  guix-build-df6dccde8e1c/output/powerpc64le-linux-gnu/SHA256SUMS.part
bdb7d702ec7f0fed0b97e9abb36dc9b1359422db6ccb777b19e01a367d4b7f23  guix-build-df6dccde8e1c/output/powerpc64le-linux-gnu/bitcoin-df6dccde8e1c-powerpc64le-linux-gnu-debug.tar.gz
56b8d13bcf7e8c5e99a44b9d8716539c3fa2ffdc0c0eba1ca7132d77c4873b0b  guix-build-df6dccde8e1c/output/powerpc64le-linux-gnu/bitcoin-df6dccde8e1c-powerpc64le-linux-gnu.tar.gz
001bd8d3707983f8f497b4efc6d985530c7beaa4373710520cecf63ef20354e3  guix-build-df6dccde8e1c/output/riscv64-linux-gnu/SHA256SUMS.part
ae61d0f8343183dddbf0c0e9e284b085d2246f167ba6e54d5c63e096e1aec363  guix-build-df6dccde8e1c/output/riscv64-linux-gnu/bitcoin-df6dccde8e1c-riscv64-linux-gnu-debug.tar.gz
7a23be4b3c254bd6b808a1edf4412da78fb3d314a5df3d643422eced15d56ec6  guix-build-df6dccde8e1c/output/riscv64-linux-gnu/bitcoin-df6dccde8e1c-riscv64-linux-gnu.tar.gz
e74ab13566563fe55e3330c0832107de2f91eb073573df6fe9c8b2d51b8248e6  guix-build-df6dccde8e1c/output/x86_64-apple-darwin/SHA256SUMS.part
62f4fecf0e72d99997142e3212b8bb2e3f5010420b8ce224ab09befd015c7a15  guix-build-df6dccde8e1c/output/x86_64-apple-darwin/bitcoin-df6dccde8e1c-x86_64-apple-darwin-unsigned.dmg
0a6cc1e2d9358ecacf3b45b9a26a85a5cdddff83ed823e718cca77df692be8e1  guix-build-df6dccde8e1c/output/x86_64-apple-darwin/bitcoin-df6dccde8e1c-x86_64-apple-darwin-unsigned.tar.gz
b4767ce20907aac2d6698c23bc7ca47865d9269e61e1801a610c9642146e170d  guix-build-df6dccde8e1c/output/x86_64-apple-darwin/bitcoin-df6dccde8e1c-x86_64-apple-darwin.tar.gz
e42b4cd134a6690897064fccb6b91ea916f8c37a43d1701c835c20d414ad7b22  guix-build-df6dccde8e1c/output/x86_64-linux-gnu/SHA256SUMS.part
fe94e2e6f807a714a65032f3a4da100b847dbf89cc6e6058a09325cb889c50d1  guix-build-df6dccde8e1c/output/x86_64-linux-gnu/bitcoin-df6dccde8e1c-x86_64-linux-gnu-debug.tar.gz
011b67e13718d024202a7e38a4d31f9bd7c7f3bb2a43ee2d789db54ed35275ce  guix-build-df6dccde8e1c/output/x86_64-linux-gnu/bitcoin-df6dccde8e1c-x86_64-linux-gnu.tar.gz
bb578a81351ddbb25d947a42f7b7649ea518b98bf5dfda9a6b04904c05829f90  guix-build-df6dccde8e1c/output/x86_64-w64-mingw32/SHA256SUMS.part
d4d7a63ba0b1d6ef85bfddf6124a1214bc3c51dfe96d0a24d1bfb90c6e30b246  guix-build-df6dccde8e1c/output/x86_64-w64-mingw32/bitcoin-df6dccde8e1c-win64-debug.zip
28aaf5124d1f15c3cf2ffc0f40b0f95761e921cae13b3cf477dc00d776e664d0  guix-build-df6dccde8e1c/output/x86_64-w64-mingw32/bitcoin-df6dccde8e1c-win64-setup-unsigned.exe
4e3880f2fdb0cbee261313d3d35f14b6a189a4a81d00730961a03b5f1728399b  guix-build-df6dccde8e1c/output/x86_64-w64-mingw32/bitcoin-df6dccde8e1c-win64-unsigned.tar.gz
1edb0861de7ecfe4893b8699a1f16723a11e480f1d6a4a5f3034e895e594ba9d  guix-build-df6dccde8e1c/output/x86_64-w64-mingw32/bitcoin-df6dccde8e1c-win64.zip
```

achow101 commented:
Closing this as it has not had any activity in a while. If you are interested in continuing work on this, please leave a comment so that it can be reopened.
